File: lib/Schema/FirebaseSchema.dart
class FirebaseSchema {
  String id;
  String name;
  String email;
  Map<String, String> contacts_RoomId;
  List<dynamic> mailBox;

  FirebaseSchema({
    required this.id,
    required this.name,
    required this.email,
    Map<String, String>? contacts_RoomId,
    List<dynamic>? mailBox,
  })  : contacts_RoomId = contacts_RoomId ?? {},
        mailBox = mailBox ?? [];

  // Getters and Setters
  String get getId => id;
  set setId(String id) => this.id = id;

  String get getName => name;
  set setName(String name) => this.name = name;

  String get getEmail => email;
  set setEmail(String email) => this.email = email;

  Map<String, String> get getContacts_RoomId => contacts_RoomId;
  set setContacts_RoomId(Map<String, String> contacts_RoomId) =>
      this.contacts_RoomId = contacts_RoomId;

  List<dynamic> get getMailBox => mailBox;
  set setMailBox(List<dynamic> mailBox) => this.mailBox = mailBox;

  // Serialization
  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'name': name,
      'email': email,
      'contacts_RoomId': contacts_RoomId,
      'mailBox': mailBox,
    };
  }

  factory FirebaseSchema.fromJson(Map<String, dynamic> json) {
    return FirebaseSchema(
      id: json['id'] ?? '',
      name: json['name'] ?? '',
      email: json['email'] ?? '',
      contacts_RoomId: Map<String, String>.from(json['contacts_RoomId'] ?? {}),
      mailBox: List<dynamic>.from(json['mailBox'] ?? []),
    );
  }
}

File: src/main/java/com/example/ChatServer/Firebase/FirebaseSchema.java
package com.example.ChatServer.Firebase;


import com.example.ChatServer.MessageSchema;

import java.util.ArrayList;
import java.util.HashMap;

import java.util.List;
import java.util.Map;


public class FirebaseSchema {
    public FirebaseSchema() {
    }
    public FirebaseSchema(String id,String email){
        setId(id);
        setName("");
        setEmail(email);
        setProfileUrl("");
        setContacts_RoomId(new HashMap<>());
        setMailBox(new ArrayList<>());
    }

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public Map<String, String> getContacts_RoomId() {
        return contacts_RoomId;
    }

    public void setContacts_RoomId(Map<String, String> contacts_RoomId) {
        this.contacts_RoomId = contacts_RoomId;
    }

    private String id;
    private String name;
    private String email;
private List mailBox;
private String profileUrl;

    public String getProfileUrl() {
        return profileUrl;
    }

    public void setProfileUrl(String profileUrl) {
        this.profileUrl = profileUrl;
    }

    public List getMailBox() {
        return mailBox;
    }

    public void setMailBox(List mailBox) {
        this.mailBox = mailBox;
    }

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    private Map<String,String> contacts_RoomId;




}
